lib/blockscore/base.rb

```diff
@@ -4,19 +4,37 @@ module BlockScore
   class Base
     extend Connection
 
-    attr_reader :attributes
-
-    def initialize(options = {})
+    def initialize(options = {}, &block)
+      @loaded = !(block)
+      @proc = block
       @attributes = options
     end
 
+    def attributes
+      return @attributes if @loaded
+      force!
+      @attributes
+    end
+
+    def force!
+      res = @proc.call
+      @attributes = res.attributes.merge(@attributes)
+      @loaded = true
+      self
+    end
+
+    def id
+      @attributes.fetch(:id, nil)
+    end
+
     def inspect
-      "#<#{self.class}:0x#{object_id.to_s(16)} JSON: " + JSON.pretty_generate(attributes)
+      str_attr = "JSON:#{JSON.pretty_generate(attributes)}"
+      "#<#{self.class}:0x#{object_id.to_s(16)} #{str_attr}>"
     end
 
     def refresh
-      r = self.class.retrieve(id)
-      @attributes = r.attributes
+      res = self.class.retrieve(id)
+      @attributes = res.attributes
```

Comment:
I know you didn't write this part but these two lines could likely just be one, right?